compiler-plugin/src/main/kotlin/com/google/devtools/ksp/processing/impl/ResolverImpl.kt

```diff
@@ -581,7 +586,7 @@ class ResolverImpl(
                 resolved?.unsubstitutedPrimaryConstructor ?: resolved?.constructors?.singleOrNull()
             }
             else -> throw IllegalStateException("unexpected class: ${function.javaClass}")
-        } as FunctionDescriptor?
```

Comment:
There are more places in this file that used this style of type casting. You can clean up those in a follow-up PR.